selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/hotupdate/RecreateUpdate.java

```diff
@@ -0,0 +1,171 @@
+        .usePostMethod()
+        .request();
+
+    // make sure, that system is prepared to  shutdown and than ready to shutdown
+    checkExpectedStatusesAndWorkspaceAfterShutDowning();
+
+    // performs rollout
+    openShiftCliCommandExecutor.execute(ocClientRolloutCommand);
+    waitOpenShiftStatus(
+        timeLimitInSecForRecreateingUpdate, delayBetweenRequest, WsMasterStauses.RUNNING);
+
+    // get current version of deployment after rollout
+    int cheDeploymentAfterRollout =
+        parseInt(openShiftCliCommandExecutor.execute(COMMAND_FOR_GETTING_CURRENT_DEPLOYMENT_CHE));
+    // After rollout updating - deployment should be increased on 1. So we previews version +1
+    // should be equal current
+    assertEquals(cheDeploymentAfterRollout, cheDeploymentBeforeRollout += 1);
+
+    // make sure that CHE ide is available after updating again
+    ide.open(workspace);
+    ide.waitOpenedWorkspaceIsReadyToUse();
+  }
+
+  private void checkExpectedStatusesAndWorkspaceAfterShutDowning() throws Exception {
+    int timeLimitForReadyToShutdownStatus = 30;
+    int delayBetweenRequestes = 1;
+
+    waitOpenShiftStatus(
+        timeLimitForReadyToShutdownStatus,
+        delayBetweenRequestes,
+        WsMasterStauses.PREPARING_TO_SHUTDOWN);
+
+    waitOpenShiftStatus(
+        timeLimitForReadyToShutdownStatus,
+        delayBetweenRequestes,
+        WsMasterStauses.READY_TO_SHUTDOWN);
+
+    // reopen the workspace and make sure that this one is not available after suspending system
+    ide.open(workspace);
+    projectExplorer.waitProjectExplorerIsNotPresent(timeLimitForReadyToShutdownStatus);
+    terminal.waitTerminalIsNotPresent(timeLimitForReadyToShutdownStatus);
+  }
+
+  private void waitOpenShiftStatus(
+      int maxTimeLimitInSec, int delayBetweenRequestsInSec, WsMasterStauses status)
+      throws Exception {
+
+    // if the limit is not exceeded - do request and check status of the system
+    while (maxTimeLimitInSec > 0) {
+      boolean isStatusFinished = getCurrentRollingStatus().equals(status.toString());
+
+      if (isStatusFinished) {
+        break;
+      }
+
+      // delay if expected status has been not achieved and decrement limit
+      WaitUtils.sleepQuietly(delayBetweenRequestsInSec);
+      maxTimeLimitInSec -= delayBetweenRequestsInSec;
+
+      // if the limit has exceeded and we have not achieved expected status - something went wrong
+      if (maxTimeLimitInSec <= 0) {
+        throw new RuntimeException(
+            "The limit has passed or something went wrong with the test environment");
+      }
+    }
+  }
+
+  private String getCurrentRollingStatus() throws Exception {
+    String restUrlForGettingSuspendingStatus =
+        cheTestApiEndpointUrlProvider.get().toString() + "system/state";
+
+    // get current response code - if system is suspended,  we will get IO exception from the
+    // server. This mean that we have suspended status
+    try {
```

Comment:
I would suggest to extract system-specific methods to the separate class with name like org.eclipse.che.selenium.core.client.CheTestSystemClient